pyOCD/flash/flash.py

```diff
@@ -211,16 +211,40 @@ def loadPageBuffer(self, bufferNumber, flashPtr, bytes):
         # transfer the buffer to device RAM
         self.target.writeBlockMemoryUnaligned8(self.page_buffers[bufferNumber], bytes)
 
+    def programPhrase(self, flashPtr, bytes):
+        """
+        Flash one page
+        """
+
+        # prevent security settings from locking the device
+        bytes = self.overrideSecurityBits(flashPtr, bytes)
+
+        # first transfer in RAM
+        self.target.writeBlockMemoryUnaligned8(self.begin_data, bytes)
+
+        # update core register to execute the program_page subroutine
+        result = self.callFunction(self.flash_algo['pc_program_page'], flashPtr, len(bytes), self.begin_data)
```

Comment:
Good catch. The difference form programPage is that this will write only the number of bytes you pass in. programPage always writes the full page size as determined by page info. I guess not all devices support sub-page programming.

Maybe we should have a flash algo dict entry for that feature, as well as the minimum program length (phrase size in Freescale NVM terminology). Will add this.